submissions/SamVal007/Memory_Pair_Game/main.js

```diff
@@ -0,0 +1,134 @@
+let similarCards = [];
+let blockedClick = false;
+const images = ['Dart_Weider', 'lyuk-skajuoker', 'R2-D2', 'c-3po',
+                'printsessa-leya-organa', 'enakin-skajuoker', 'obi-van-kenobi',
+                 'dart-sidius', 'kvaj-gon-dzhinn','joda'];
+
+const mainAttributes = {
+    container: 'container',
+    game: 'game',
+    card: 'card',
+    front: 'front_card',
+    frontImg: 'img-front_card',
+    back: 'back_card',
+    backImg: 'img-back_card',
+    flipped: 'flipped',
+    hidden: 'hidden'
+};
+
+const cleanCountArray = () => similarCards = [];
+
+const flippSelectedCards = () => {
+    setTimeout(() => {
+        document.querySelectorAll(".flipped").forEach((card) => {
+            card.classList.remove(mainAttributes.flipped);
+            cleanCountArray();
+            blockedClick = false;
+        });
+    }, 900);
+};
+
+const hideSimilarCards = () => {
```

Comment:
[Separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns)

this function not just hide cards, it also check win condition

Reply:
@SamVal007 not fixed

Reply:
Yep, sorry, It's still not fixed.

Reply:
So, I will put some my toughs here if it's ok:
1) Firs what did I made a function for checking win like this:
const checkingForTheVictory = () => {
    const counterHiddenCards = document.querySelectorAll(".hidden").length;
    console.log(counterHiddenCards);
    const amountOfCards = ((images.length - 1) * enoughCountOfSimilarCards);
    console.log(amountOfCards);
    if (counterHiddenCards === amountOfCards) {
        victoryMessage();
    }
}

Reply:
2. I was try to use in front of func hideSimilarCards, but the problem is that  "hideSimilarCards" is asynchronous function and whehever I put func   "checkingForTheVictory" - It will be done faster then "hideSimilarCards" and condition would be not correct!

Reply:
So I have two points: 
1) To put "checkingForTheVictory" inside of "hideSimilarCards"
2) Change  this fucn like this:
const checkingForTheVictory = () => {
    const counterHiddenCards = document.querySelectorAll(".hidden").length;
    console.log(counterHiddenCards);
    const amountOfCards = **((images.length - 1) * enoughCountOfSimilarCards);**
    console.log(amountOfCards);
    if (counterHiddenCards === amountOfCards) {
        victoryMessage();
    }
}

Reply:
Ah, Yeah 3rd way is to use Promise

I'm not sure which way is better...

Reply:
main point was that this has naming hideCards, but do not only hiding. Change naming to more obvious () or made some smaller functions  like in example below and combine in one bigger (i prefer second way)
```
const hideCards = () => {
  // hide logic without set timeout, cleanCountArray, checkingForTheVictory
}
```

Reply:
@SamVal007